.circleci/config/jobs/operator_upgrade_tests.yml

```diff
@@ -0,0 +1,220 @@
+description: |
+  Deploys a previously released version of the snyk-operator.
+  Subsequently upgrades the Operator with a new version that is intended
+  to be released. If the Operator reaches the running state in both cases,
+  we can aassume that it's able to upgrade.
+
+executor: redhat-openshift/default
+
+working_directory: ~/kubernetes-monitor
+
+steps:
+  - checkout
+
+  - run:
+      name: Install required packages
+      command: |
+        sudo apt install -y uuid-runtime
+        python -m pip install requests pyyaml
+        python -m pip install operator-courier
+
+  - run:
+      name: Update /etc/hosts with an entry to the OpenShift cluster
+      description: |
+        The test cluster returns URLs that are local only to the cluster (e.g. https://api.crc.testing).
+        These URLs don't make sense for the public internet, so we add a little hack in /etc/hosts 
+        that makes the URLs point to the IP address of the test cluster, allowing us to reach back to it.
+      command: echo "${OPENSHIFT4_ETC_HOSTS_ENTRY}" | sudo tee -a /etc/hosts
+
+  - redhat-openshift/login-and-update-kubeconfig:
```

Comment:
It's used in all subsequent steps because we run `kubectl` against the OS4 test cluster, the login + updating KUBECONFIG helps point kubectl to the cluster and to have the auth token